File: src/mysqlcookbook/recipes/api/04_stmt/placeholder_list.py
#!/usr/bin/python3
# placeholder_list.py

# Create a string of placeholder characters, with one ? character
# per element in an array of values.

# Array of values

values = [1, 2, 3, 4, 5]

# Method 1: use * to "multiply" and array and join to put commas between
# elements. Note that join requires a list of _strings_.

#@ _JOIN_METHOD_
str = ",".join(["?"] * len(values))
#@ _JOIN_METHOD_

print("Method 1: %s" % str)

#@ _CONCAT_METHOD_
str = ""
if len(values) > 0:
  str = "?"
for i in range(1, len(values)):
  str += ",?"
#@ _CONCAT_METHOD_

print("Method 2: %s" % str)

File: src/mysqlcookbook/recipes/api/04_stmt/placeholder_list.php
<?php
# placeholder_list.php

# Create a string of placeholder characters, with one ? character
# per element in an array of values.

# Array of values

$values = array (1, 2, 3, 4, 5);

#@ _CONCAT_METHOD_
$str = "";
if (count ($values))
  $str = "?";
for ($i = 1; $i < count ($values); $i++)
  $str .= ",?";
#@ _CONCAT_METHOD_

print ("$str\n");
?>
